### PR TITLE
Relocate openfermion imports and extract PySCF dependency

### PR DESCRIPTION
**Pull request by agran2018**
WIP PR. 
- Relocate OpenFermion import statements to reduce the time to import PennyLane.
- Extract PySCF dependency from PennyLane.

### CODE REVIEW

File: qchem/pennylane_qchem/qchem/obs.py [hunk 1]
```diff
@@ -192,6 +190,8 @@ def spin2(electrons, orbitals, mapping="jordan_wigner", wires=None):
     + (0.125) [Xw0 Yw1 Xw2 Yw3]
     """
 
+    from openfermion.ops import FermionOperator
```
Comment (josh146) [inline qchem/pennylane_qchem/qchem/obs.py:193]: @agran2018, this import should be okay to leave at the top of the file?

File: qchem/pennylane_qchem/qchem/obs.py [hunk 2]
```diff
@@ -295,6 +295,9 @@ def observable(fermion_ops, init_term=0, mapping="jordan_wigner", wires=None):
     + (-0.075) [Z0 Z2]
     """
 
+    from openfermion.ops import FermionOperator
+    from openfermion.transforms import bravyi_kitaev, jordan_wigner
```
Comment (josh146) [inline qchem/pennylane_qchem/qchem/obs.py:299]: Same here

File: qchem/pennylane_qchem/qchem/structure.py [hunk 3]
```diff
@@ -18,10 +18,6 @@
 from shutil import copyfile
 
 import numpy as np
-from openfermion import MolecularData, QubitOperator
-from openfermion.transforms import bravyi_kitaev, get_fermion_operator, jordan_wigner
-from openfermionpsi4 import run_psi4
-from openfermionpyscf import run_pyscf
```
Comment (josh146) [inline qchem/pennylane_qchem/qchem/structure.py:24]: @agran2018 it's my understanding that `run_psi4` and `run_pyscf` should be imported within the scf driver functions, but the other OpenFermion imports should be okay to leave at the top of the file

**Comment by agran2018** [comment]
Final solution implemented in PR #1302

**Pull request closed without merging**